### PR TITLE
Manage the gke build cluster via argocd - part one

### PR DESCRIPTION
**Pull request by upodroid**
This PR configures argocd to manage the test-pods namespace of the gke build cluster except the secrets. We need to migrate the secrets to ESO from KES and resolve https://github.com/kubernetes/k8s.io/issues/3123 in a separate PR.

### CODE REVIEW

File: kubernetes/gke-utility/argocd/clusters.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ metadata:
     argocd.argoproj.io/secret-type: cluster
     clusterType: prow
     environment: prod
+    prow-namespace: default
```
Comment (xmudrii) [inline kubernetes/gke-utility/argocd/clusters.yaml:9]: Very tiny nit: maybe go with `prowNamespace` to match other variables, mainly `clusterType`

File: kubernetes/gke-utility/argocd/clusters.yaml [hunk 2]
```diff
@@ -23,3 +24,30 @@ stringData:
       }
     }
 ---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: gke-prow-build
```
Comment (xmudrii) [inline kubernetes/gke-utility/argocd/clusters.yaml:30]: What's this used for?

File: kubernetes/gke-utility/argocd/clusters.yaml (hunk 1 above)
Comment (upodroid) [reply kubernetes/gke-utility/argocd/clusters.yaml]: Fixed

File: kubernetes/gke-utility/argocd/clusters.yaml (hunk 2 above)
Comment (xmudrii) [reply kubernetes/gke-utility/argocd/clusters.yaml:30]: @upodroid I was wondering what that kubeconfig is used, I think I might asked the same in the previous PR but forgot 😅

File: kubernetes/gke-utility/argocd/clusters.yaml [hunk 3]
```diff
@@ -23,3 +24,30 @@ stringData:
...
+  labels:
+    argocd.argoproj.io/secret-type: cluster
+    clusterType: prow
+    environment: prod
+    prow-namespace: test-pods
```
Comment (xmudrii) [inline kubernetes/gke-utility/argocd/clusters.yaml:35]: Should we also change that to `prowNamespace`?

Comment (upodroid) [reply kubernetes/gke-utility/argocd/clusters.yaml]: Oops, fixed

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**